shell/platform/android/io/flutter/embedding/engine/systemchannels/RestorationChannel.java

```diff
@@ -0,0 +1,111 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.systemchannels;
+
+import android.support.annotation.NonNull;
+import io.flutter.Log;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.StandardMethodCodec;
+
+/**
+ * System channel to exchange restoration data between framework and engine.
+ *
+ * <p>The engine can obtain the current restoration data from the framework via this channel to
+ * store it on disk and - when the app is relaunched - provide the stored data back to the framework
+ * to recreate the original state of the app.
```

Comment:
can you describe the timing a bit more? data has to be set before being read from the framework and before being read from the engine. saves can be set many times and read x times. restores can only be set one time and read one time.

Reply:
Expanded the doc.